### ISSUE TITLE
Add health for Infinispan to the health endpoint

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#32781]
### Description

Once you start more than one instance of Keycloak and start enabling clustering using JGroups, it would be good to monitor that the discovery worked, and that at least one other node was discovered.

### Motivation

If the nodes don't discover each other, bad things can happen (outdated realm information, missing authentication sessions, etc.)

We could add a parameter of a minimum number of nodes to be discovered to report the instance as ready. This would handle split-brain scenarios. In a deployment of 3 instances, at least 2 instances will need to discover each other to form a majority.

### Details

`KeyloakSession` can be injected using `@Inject` in health endpoints, and the session factory can be picked up using `QuarkusKeycloakSessionFactory.getInstance()`

**Comment by vmuzikar** [comment]
> So it might actually be a generic SPI, or interface to be added to provider factories (similar to `ServerInfoAwareProviderFactory`).

Yes, this would make sense. Thank you for opening this RFE, @ahus1! Added it to the Cloud Native's backlog.

**Comment by ahus1** [comment]
With JDBC Ping, all nodes talk to the database. And we have a split-brain detection in place that only one slice of the cluster will mark itself ready. It is also part of the readiness probe. 

This should resolve the original need. Closing.
